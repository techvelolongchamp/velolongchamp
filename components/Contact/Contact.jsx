import React from 'react'
import { IoLogoFacebook, IoLogoTwitter, IoLogoInstagram } from 'react-icons/io'

import { ContactWrapper, SocialWrapper } from './Contact.styled'

const Contact = () => {
  return (
    <ContactWrapper>
      <img src="/logo.png" alt="logo" />
      <SocialWrapper>
        <a
          href="https://www.facebook.com/AssoVeloLongchamp"
          title="Facebook"
          target="_blank"
          rel="noreferrer"
          className="facebook"
        >
          <IoLogoFacebook />
        </a>
        <a
          href="https://twitter.com/velolongchamp"
          title="Twitter"
          target="_blank"
          rel="noreferrer"
          className="twitter"
        >
          <IoLogoTwitter />
        </a>
        <a
          href="https://www.instagram.com/velolongchamp/"
          title="Instagram"
          target="_blank"
          rel="noreferrer"
          className="instagram"
        >
          <IoLogoInstagram />
        </a>
      </SocialWrapper>
      <p>
        <a
          href="mailto:contact@velo-longchamp.fr?subject=Contact"
          title="Contact"
        >
          contact@velo-longchamp.fr
        </a>
      </p>
    </ContactWrapper>
  )
}

export default Contact
