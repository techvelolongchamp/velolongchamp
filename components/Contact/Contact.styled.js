import styled from 'styled-components'

export const ContactWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: ${({ theme }) => theme.spacing.xl};

  img {
    width: 200px;
    padding: 10px 0;
  }

  p {
    font-size: ${({ theme }) => theme.fonts.medium};
  }
  a {
    color: inherit;
  }
`

export const SocialWrapper = styled.div`
  svg {
    width: 30px;
    height: 30px;
    margin: 0 5px;
  }

  .facebook {
    color: #3b5998;
  }

  .twitter {
    color: #00acee;
  }
`
