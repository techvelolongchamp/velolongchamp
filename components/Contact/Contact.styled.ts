import styled from 'styled-components'

export const ContactWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: ${({ theme }) => theme.spacing.xl};

  p {
    font-size: ${({ theme }) => theme.fonts.medium};
  }
  a {
    color: inherit;
  }
`

export const SocialWrapper = styled.div`
  padding-top: 10px;
  svg {
    width: 30px;
    height: 30px;
    margin: 0 5px;
  }

  .facebook {
    color: #3b5998;
  }

  .twitter {
    color: #00acee;
  }
`
