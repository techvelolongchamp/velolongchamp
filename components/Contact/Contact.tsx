import React from 'react'
import Image from 'next/image'
import {
  IoLogoFacebook,
  IoLogoTwitter,
  IoLogoInstagram,
  IoLogoYoutube,
} from 'react-icons/io'

import { ContactWrapper, SocialWrapper } from './Contact.styled'

const Contact = () => {
  return (
    <ContactWrapper>
      <Image src="/logo.png" alt="logo" width={200} height={150} />
      <SocialWrapper>
        <a
          href="https://www.facebook.com/AssoVeloLongchamp"
          title="Facebook"
          target="_blank"
          rel="noreferrer"
          className="facebook"
        >
          <IoLogoFacebook />
        </a>
        <a
          href="https://twitter.com/velolongchamp"
          title="Twitter"
          target="_blank"
          rel="noreferrer"
          className="twitter"
        >
          <IoLogoTwitter />
        </a>
        <a
          href="https://www.instagram.com/velolongchamp/"
          title="Instagram"
          target="_blank"
          rel="noreferrer"
          className="instagram"
        >
          <IoLogoInstagram />
        </a>
        <a
          href="https://www.youtube.com/channel/UCILGPS4lvBE5GXbTnFze5rA"
          title="Youtube"
          target="_blank"
          rel="noreferrer"
          className="youtube"
        >
          <IoLogoYoutube />
        </a>
      </SocialWrapper>
      <p>
        <a
          href="mailto:contact@velo-longchamp.fr?subject=Contact"
          title="Contact"
        >
          contact@velo-longchamp.fr
        </a>
      </p>
    </ContactWrapper>
  )
}

export default Contact
