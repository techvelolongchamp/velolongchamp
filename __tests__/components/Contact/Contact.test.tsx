import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import Contact from '../../../components/Contact/Contact'

describe('Contact unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<Contact />)

    expect(container).toMatchSnapshot()
  })
})
