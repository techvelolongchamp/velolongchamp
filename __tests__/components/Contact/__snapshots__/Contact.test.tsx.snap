// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact unit tests should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 30px;
}

.c0 p {
  font-size: 14px;
}

.c0 a {
  color: inherit;
}

.c1 {
  padding-top: 10px;
}

.c1 svg {
  width: 30px;
  height: 30px;
  margin: 0 5px;
}

.c1 .facebook {
  color: #3b5998;
}

.c1 .twitter {
  color: #00acee;
}

.c1 .youtube {
  color: #ff0000;
}

.c1 .instagram {
  color: #c13584;
}

<div>
  <div
    class="c0"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="logo"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <div
      class="c1"
    >
      <a
        class="facebook"
        href="https://www.facebook.com/AssoVeloLongchamp"
        rel="noreferrer"
        target="_blank"
        title="Facebook"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M426.8 64H85.2C73.5 64 64 73.5 64 85.2v341.6c0 11.7 9.5 21.2 21.2 21.2H256V296h-45.9v-56H256v-41.4c0-49.6 34.4-76.6 78.7-76.6 21.2 0 44 1.6 49.3 2.3v51.8h-35.3c-24.1 0-28.7 11.4-28.7 28.2V240h57.4l-7.5 56H320v152h106.8c11.7 0 21.2-9.5 21.2-21.2V85.2c0-11.7-9.5-21.2-21.2-21.2z"
          />
        </svg>
      </a>
      <a
        class="twitter"
        href="https://twitter.com/velolongchamp"
        rel="noreferrer"
        target="_blank"
        title="Twitter"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M492 109.5c-17.4 7.7-36 12.9-55.6 15.3 20-12 35.4-31 42.6-53.6-18.7 11.1-39.4 19.2-61.5 23.5C399.8 75.8 374.6 64 346.8 64c-53.5 0-96.8 43.4-96.8 96.9 0 7.6.8 15 2.5 22.1-80.5-4-151.9-42.6-199.6-101.3-8.3 14.3-13.1 31-13.1 48.7 0 33.6 17.2 63.3 43.2 80.7-16-.4-31-4.8-44-12.1v1.2c0 47 33.4 86.1 77.7 95-8.1 2.2-16.7 3.4-25.5 3.4-6.2 0-12.3-.6-18.2-1.8 12.3 38.5 48.1 66.5 90.5 67.3-33.1 26-74.9 41.5-120.3 41.5-7.8 0-15.5-.5-23.1-1.4C62.8 432 113.7 448 168.3 448 346.6 448 444 300.3 444 172.2c0-4.2-.1-8.4-.3-12.5C462.6 146 479 129 492 109.5z"
          />
        </svg>
      </a>
      <a
        class="instagram"
        href="https://www.instagram.com/velolongchamp/"
        rel="noreferrer"
        target="_blank"
        title="Instagram"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M336 96c21.2 0 41.3 8.4 56.5 23.5S416 154.8 416 176v160c0 21.2-8.4 41.3-23.5 56.5S357.2 416 336 416H176c-21.2 0-41.3-8.4-56.5-23.5S96 357.2 96 336V176c0-21.2 8.4-41.3 23.5-56.5S154.8 96 176 96h160m0-32H176c-61.6 0-112 50.4-112 112v160c0 61.6 50.4 112 112 112h160c61.6 0 112-50.4 112-112V176c0-61.6-50.4-112-112-112z"
          />
          <path
            d="M360 176c-13.3 0-24-10.7-24-24s10.7-24 24-24c13.2 0 24 10.7 24 24s-10.8 24-24 24zM256 192c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64m0-32c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96z"
          />
        </svg>
      </a>
      <a
        class="youtube"
        href="https://www.youtube.com/channel/UCILGPS4lvBE5GXbTnFze5rA"
        rel="noreferrer"
        target="_blank"
        title="Youtube"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M508.6 148.8c0-45-33.1-81.2-74-81.2C379.2 65 322.7 64 265 64h-18c-57.6 0-114.2 1-169.6 3.6C36.6 67.6 3.5 104 3.5 149 1 184.6-.1 220.2 0 255.8c-.1 35.6 1 71.2 3.4 106.9 0 45 33.1 81.5 73.9 81.5 58.2 2.7 117.9 3.9 178.6 3.8 60.8.2 120.3-1 178.6-3.8 40.9 0 74-36.5 74-81.5 2.4-35.7 3.5-71.3 3.4-107 .2-35.6-.9-71.2-3.3-106.9zM207 353.9V157.4l145 98.2-145 98.3z"
          />
        </svg>
      </a>
    </div>
    <p>
      <a
        href="mailto:contact@velo-longchamp.fr?subject=Contact"
        title="Contact"
      >
        contact@velo-longchamp.fr
      </a>
    </p>
  </div>
</div>
`;
